src/PKSim.Core/Services/TransportContainerUpdater.cs

```diff
@@ -15,61 +18,62 @@ public interface ITransportContainerUpdater
       ///    Update the default transporter settings using the template defined in the database based on the
       ///    <paramref name="transporterName" />/>
       /// </summary>
-      void SetDefaultSettingsForTransporter(ISimulationSubject simulationSubject, IndividualTransporter transporter, string transporterName) ;
+      void SetDefaultSettingsForTransporter(ISimulationSubject simulationSubject, IndividualTransporter transporter, string transporterName);
 
-      void SetDefaultSettingsForTransporter(ISimulationSubject simulationSubject, IndividualTransporter transporter, TransportType transportType) ;
+      void SetDefaultSettingsForTransporter(ISimulationSubject simulationSubject, IndividualTransporter transporter, TransportType transportType);
    }
 
    public class TransportContainerUpdater : ITransportContainerUpdater
    {
       private readonly ITransporterContainerTemplateRepository _transporterContainerTemplateRepository;
+      private readonly ITransporterTemplateRepository _transporterTemplateRepository;
       private readonly IEventPublisher _eventPublisher;
 
       public TransportContainerUpdater(
          ITransporterContainerTemplateRepository transporterContainerTemplateRepository,
+         ITransporterTemplateRepository transporterTemplateRepository,
          IEventPublisher eventPublisher)
       {
          _transporterContainerTemplateRepository = transporterContainerTemplateRepository;
+         _transporterTemplateRepository = transporterTemplateRepository;
          _eventPublisher = eventPublisher;
       }
 
       public void SetDefaultSettingsForTransporter(ISimulationSubject simulationSubject, IndividualTransporter transporter, string transporterName)
       {
          var speciesName = simulationSubject.Species.Name;
-         transporter.TransportType = _transporterContainerTemplateRepository.TransportTypeFor(speciesName, transporterName);
+         var transportType = _transporterTemplateRepository.TransportTypeFor(speciesName, transporterName);
+         transporter.TransportType = transportType;
+
+         if (!_transporterTemplateRepository.HasTransporterTemplateFor(speciesName, transporterName))
+         {
+            //No template was found for the given name. Raise event warning
+            _eventPublisher.PublishEvent(new NoTransporterTemplateAvailableEvent(transporter));
+            return;
+         }
```

Comment:
why are the lines 45-46 called BEFORE the lines 48-53?

Reply:
Well there is still a default being returned if not found. in the TemplateRepository. ,Maywe we should rename to TransportTypeOrDefault

Reply:
yep, would be good to rename it, otherwise the code here looking strange (without going in the called functions)

Reply:
ok,  just created some additional issues.
If you rename TransportTypeFor=>TransportTypeOrDefault we can merge I think